### PR TITLE
delete verbose clone, add cita 0.23

### PR DESCRIPTION
**Pull request by clearloop**
## Optimized patch and version 0.23

+ delete the verbose clone about `env.sh`
+ add `cita@0.23` formula

About the #5 issue, cita before 0.22 does not match the corresponding `env.sh`, if we install them with formula, it will be a lot of work to do = =

I saw some formula can merge versions in just one `.rb` file, but I forget which it is, so I just use different files like [Formula/ruby](https://github.com/Homebrew/homebrew-core/search?q=ruby&unscoped_q=ruby). 

I will pr the formula to the cleanest type (if one file is?) any time I find the solution.

## test 

```shell
$ brew tap clearloop/cita
$ brew install cita@0.23
# brew install cita
```

### CODE REVIEW

File: cita@0.23.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# coding: utf-8
+# CITA Formula
+class CitaAT023 < Formula
+  desc "A high performance blockchain for enterprise users."
+  homepage "https://github.com/cryptape/cita"
+
+  version "0.23.1"
```
Comment (leeyr338) [inline cita@0.23.rb:7]: Is it better to install `0.23.0` when our user:

```shell
brew install cita@0.23
```

?

**Comment by leeyr338** [comment]
> I saw some formula can merge versions in just one .rb file, but I forget which it is, so I just use different files like Formula/ruby.

It may be better to use different files for different version. 
@kaikai1024 , what do you think?

> About the #5 issue, cita before 0.22 does not match the corresponding env.sh, if we install them with formula, it will be a lot of work to do = =

I think it is okay for CITA supports formula from v0.22.0 .

File: cita@0.23.rb (hunk 1 above)
Comment (kaikai1024) [reply cita@0.23.rb:7]: i don't think so.

We can just focus the minor version and use the latest patch version:

cita@0.23:  the version of CITA is 0.23.x (x will be the latest patch version)

**Comment by kaikai1024** [comment]
> It may be better to use different files for different version.

We can support the different version first. then as @clearloop said:

> I will pr the formula to the cleanest type (if one file is?) any time I find the solution.

I don't know much about  formula, we can follow the standard(if has the standard).

personaly, i don't like duplicate codes. I prefer to using one file.

This pr seems ok for me.

**Comment by kaikai1024** [comment]
Thanks for posting the test result.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+**/*.DS_Store
```
Comment (kaikai1024) [inline .gitignore]: EOF

File: cita@0.23.rb (hunk 1 above)
Comment (leeyr338) [reply cita@0.23.rb:7]: This is also very reasonable!

File: .gitignore (hunk 2 above)
Comment (clearloop) [reply .gitignore]: `.DS_Store` is a kind of __'junk file'__ on MacOS, it generates out of developers' control...and, [cryptape/cita](https://github.com/cryptape/cita/blob/bcc0843e7011a03de2de289f6952c8826c1959aa/.gitignore#L33) also add `.DS_Store` into `.gitignore`, here is the [explanation](https://stackoverflow.com/questions/30246832/is-ds-store-file-important).

**Comment by clearloop** [comment]
> I saw some formula can merge versions in just one .rb file, but I forget which it is, so I just use different files like Formula/ruby.

Perhaps I was in dream that time....It is said: 

> You can add a stable do url (a url outside stable do is implied to be stable), devel do url and your head URL as you currently have. Formulae do not support multiple versions. For your system you would need to have multiple formulae e.g. buckaroo and buckaroo@0.1 —— [discourse.brew.sh](https://discourse.brew.sh/t/how-do-i-add-multiple-versions-to-a-formula-on-github/910/5)

we can:
```rb
stable do
  url "..."
  sha256 "..."
end
# brew install cita
# brew install cita --stable

devel do
  url "..."
  sha256 "..."
end
# brew install cita --devel
```

but not: 
```rb
version "0.24" do
...
end

version "0.25" do
...
end
```

It seems not using duplicate files is impossible? Can't believe it!

File: .gitignore (hunk 2 above)
Comment (kaikai1024) [reply .gitignore]: yeah, i mean add a line at the end of file.
check the tips of github:

> No new line at end of file.

https://en.wikipedia.org/wiki/End-of-file

Comment (kaikai1024) [reply .gitignore]: check this:

https://github.com/cryptape/cita/blob/develop/.editorconfig#L9

Comment (clearloop) [reply .gitignore]: deal, `force-push` or `squash-merge`...

**Comment by kaikai1024** [comment]
that's ok.

It seems a good idea that support stable and dev version.

**Review by kaikai1024 — APPROVED** [review]

**Pull request merged**